### PR TITLE
provider/cpu_utils:move function to provider.cpu_utils

### PR DESCRIPTION
**Pull request by nanliu-r**
ID:1879376

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
@huangyum @vivianQizhu please help to review the code firstly, will update all test results.

Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_model.host: PASS (108.20 s)

**Comment by nanliu-r** [comment]
Test PASS.

python3 ConfigTest.py --guestname=RHEL.8.3.0,Win10 --platform=x86_64 --clone=no --testcase=x86_cpu_model,x86_cpu_flags

RESULTS    : PASS 13 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 38

File: qemu/tests/x86_cpu_model.py [hunk 1]
```diff
@@ -2,37 +2,11 @@
...
-        no_flags = params.get("no_flags")
-        if no_flags:
-            err_flags = [f for f in no_flags.split() if f in out]
-            if err_flags:
-                test.fail("Flag %s should not be present in guest" % err_flags)
-    else:
-        error_context.context("Check cpu flags on host", logging.info)
-        if missing:
-            test.cancel("This host doesn't support flag %s" % missing)
+from provider.cpu_utils import check_flags
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_model.py:8]: import order: build-in ->avocado -> virttest -> tp-qemu, so move this line down.

File: provider/cpu_utils.py [hunk 2]
```diff
@@ -99,3 +102,32 @@ def check_guest_cpu_topology(session, os_type, cpuinfo):
...
+    :param session: guest session
+    """
+    cmd = params["check_flag_cmd"]
+    func = process.getoutput
+    if session:
+        func = session.cmd_output
+    out = func(cmd).split()
+    missing = [f for f in flags.split() if f not in out]
+    if session:
+        error_context.context("Check cpu flags inside guest", logging.info)
```
Comment (vivianQizhu) [inline provider/cpu_utils.py:122]: Directly using `logging.info()` if not `@error_context` for the function.

File: qemu/tests/x86_cpu_model.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_model.py]: > import order: build-in ->avocado -> virttest -> tp-qemu, so move this line down.

update and thanks.

File: provider/cpu_utils.py (hunk 2 above)
Comment (nanliu-r) [reply provider/cpu_utils.py]: > Directly using `logging.info()` if not `@error_context` for the function.

update and thanks.

**Review by vivianQizhu — APPROVED** [review]

File: provider/cpu_utils.py [hunk 3]
```diff
@@ -99,3 +101,32 @@ def check_guest_cpu_topology(session, os_type, cpuinfo):
         logging.debug("CPU infomation of guest:\n%s", out)
 
     return is_matched
+
+
+def check_flags(params, flags, test, session=None):
```
Comment (huangyum) [inline provider/cpu_utils.py:106]: How about rename it to check_cpu_flags? Others look good to me.

Comment (nanliu-r) [reply provider/cpu_utils.py]: > How about rename it to check_cpu_flags? Others look good to me.

@huangyum  update, thanks.

Comment (huangyum) [reply provider/cpu_utils.py]: Would you pls provide the latest test results? Thanks.

Comment (nanliu-r) [reply provider/cpu_utils.py]: > Would you pls provide the latest test results? Thanks.

@huangyum update, please help to check, thanks.
 Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_flags.intel.memory_protection_key.q35: PASS (95.71 s)
 Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_model.Skylake-Client-noTSX.q35: PASS (58.88 s)

**Review by huangyum — APPROVED** [review]
ACK

**Pull request merged**